### PR TITLE
 🔧 [open-zaak/open-zaak#1629] Refactor settings module

### PR DESCRIPTION
**Pull request by stevenbal**
issue: open-zaak/open-zaak#1629

Requires https://github.com/maykinmedia/open-api-framework/pull/35 to be merged and released

### CODE REVIEW

File: src/nrc/conf/includes/base.py [hunk 1]
```diff
@@ -144,326 +27,50 @@
...
+    "level": LOG_LEVEL,
+    "propagate": False,
 }
 
-LOG_OUTGOING_REQUESTS_DB_SAVE = config("LOG_OUTGOING_REQUESTS_DB_SAVE", default=False)
+# TODO should be part of OAF?
 LOG_OUTGOING_REQUESTS_DB_SAVE_BODY = config(
     "LOG_OUTGOING_REQUESTS_DB_SAVE_BODY", default=False
 )
+# TODO should be part of OAF?
```
Comment (stevenbal) [inline src/nrc/conf/includes/base.py:58]: I was considering adding this to OAF (it's hardcoded there currently), but the default is `True` there (https://github.com/maykinmedia/open-api-framework/blob/main/open_api_framework/conf/base.py#L653). Do you think it's fine to use default `True` for Open Notificaties as well?

File: src/nrc/conf/includes/base.py [hunk 2]
```diff
@@ -144,326 +27,50 @@
...
+    "backupCount": 10,
+}
+LOGGING["loggers"]["nrc.api.serializers"] = {
+    "handlers": ["notifications"],
+    "level": LOG_LEVEL,
+    "propagate": False,
 }
 
-LOG_OUTGOING_REQUESTS_DB_SAVE = config("LOG_OUTGOING_REQUESTS_DB_SAVE", default=False)
+# TODO should be part of OAF?
```
Comment (stevenbal) [inline src/nrc/conf/includes/base.py:54]: @annashamray I was considering adding these two settings to OAF (one of them is hardcoded there currently and the other not defined), but the defaults are `True` there (https://github.com/maykinmedia/open-api-framework/blob/main/open_api_framework/conf/base.py#L653). Do you think it's fine to use default `True` for these settings for Open Notificaties as well?

File: src/nrc/conf/includes/base.py [hunk 3]
```diff
@@ -472,45 +79,10 @@
 ##############################
 
 # Django-axes
-AXES_CACHE = "axes"  # refers to CACHES setting
+# TODO these settings are different from OAF
```
Comment (stevenbal) [inline src/nrc/conf/includes/base.py:82]: In OAF failure limit is 10 and cooloff time 1. Do we want to adapt these stricter settings instead?

File: src/nrc/conf/includes/base.py (hunk 2 above)
Comment (annashamray) [reply src/nrc/conf/includes/base.py:54]: I don't see it as a problem. Having the same default for all components is a good thing. But please add a warning to the changelog that in the future version the default for this setting is changed

File: src/nrc/conf/includes/base.py (hunk 3 above)
Comment (annashamray) [reply src/nrc/conf/includes/base.py:82]: To be honest I have no opinion here. I think OAF settings are fine, but either way is ok

**Review by annashamray — APPROVED** [review]

File: src/nrc/conf/includes/base.py (hunk 2 above)
Comment (stevenbal) [reply src/nrc/conf/includes/base.py:54]: added it to the changelog

**Pull request merged**